### PR TITLE
fix: made social media links a grouped list

### PR DESCRIPTION
**Pull request by medhatnar**
### Overview

Updating SocialMediaSharing group component to be an unordered list with SocialMediaLink as list items.

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: [DISC-1122](https://codecademy.atlassian.net/browse/DISC-1122)
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] This PR includes testing instructions tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions
_Gamut instructions_
1. Go to[ /?path=/docs/brand-labs-molecules-socialmediasharing--social-media-sharing](https://6435b2fb2d1c67368f07c34c--gamut-preview.netlify.app/?path=/docs/brand-labs-molecules-socialmediasharing--social-media-sharing)
2. Using VoiceOver
2. Use CONTROL + OPTION + RIGHT ARROW keys to navigate the Social media icons(Facebook, LinkedIn & twitter)
3. Voiceover should announce that the user is in a list
4. Inspect to ensure that the social media icons are individual `li` elements within `ul` element

_Monolith Instructions_
### learner story
1. Go to a learner story here: https://pr-35303-monolith.dev-eks.codecademy.com/learner-stories/web-development/5QNjwhzuCwfMfSPezJhWra
2. Social media sharing component should be 1-1 with what is in production

_Left is update, right is production_
Screen Shot 2023-04-13 at 12 53 04 PM

### sorting quiz
1. Log in
2. Go to [explore/sorting-quiz/results](https://pr-35303-monolith.dev-eks.codecademy.com/explore/sorting-quiz/results?advocator=38&connector=44&questioner=40&solver=37)
3. Social media sharing component should be 1-1 with what is in production

_Left is update, right is production_
Screen Shot 2023-04-13 at 12 54 42 PM

### 30 day challenge confirmation modal
1. Log in
2. Go to [/30daychallenge/reminders](https://pr-35303-monolith.dev-eks.codecademy.com/30daychallenge/reminders)
3. Social media sharing component should be 1-1 with what is in production

_Left is update, right is production_
Screen Shot 2023-04-13 at 1 45 56 PM

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/35303)  | [Monolith Env](https://pr-35303-monolith.dev-eks.codecademy.com/catalog?PR_ENV=pr-35303-portal) |

[DISC-1122]: https://codecademy.atlassian.net/browse/DISC-1122?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.303-alpha.7d96e8.0<br/>@codecademy/gamut-labs@35.11.1-alpha.7d96e8.0<br/>@codecademy/styleguide@64.1.2-alpha.7d96e8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://642caa62951d590a100eb894--gamut-preview.netlify.app>https://642caa62951d590a100eb894--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/642caa62951d590a100eb894>Deploy Logs</a></p>

File: packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { GamutIconProps } from '@codecademy/gamut-icons';
 import styled from '@emotion/styled';
 import * as React from 'react';
 
-const IconLink = styled(Box)``.withComponent(Anchor);
+const IconLink = styled(Anchor)``;
```
Comment (medhatnar) [inline packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx:6]: removing `Box` component from styled definition

File: packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx [hunk 2]
```diff
@@ -39,25 +39,27 @@ export const SocialShareIconLink: React.FC<SocialShareIconLinkProps> = ({
...
-      borderRadius="50%"
-    >
-      <IconComponent
-        title={id}
-        titleId={getTitleId(id, sectionId)}
-        size={size === 'small' ? 16 : 24}
-        color="text"
-      />
-    </IconLink>
+    <Box as="li">
```
Comment (medhatnar) [inline packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx:42]: Included `Box` component in here instead so it maintains original shape and we can explicitly define it as an `li`

File: packages/gamut-labs/src/SocialMediaSharing/index.tsx [hunk 3]
```diff
@@ -100,11 +105,12 @@ export const SocialMediaSharing: React.FC<SocialMediaSharingProps> = ({
           {label}
         </Text>
       )}
-      <GridBox
+      <UnstyledUnorderedList
```
Comment (medhatnar) [inline packages/gamut-labs/src/SocialMediaSharing/index.tsx:108]: updated this `Gridbox` to be as `ul`

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.304-alpha.907bd9.0<br/>@codecademy/gamut-labs@35.11.2-alpha.907bd9.0<br/>@codecademy/styleguide@64.1.3-alpha.907bd9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://642ded374143e80890f2ca0d--gamut-preview.netlify.app>https://642ded374143e80890f2ca0d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/642ded374143e80890f2ca0d>Deploy Logs</a></p>

**Comment by zippyzow** [comment]
@medhatnar could you include testing instructions for this?

**Comment by christian-dinh** [comment]
making the `GridBox` a `ul` adds some default padding and margin from the user agent styles:

Screen Shot 2023-04-06 at 12 52 50 PM

**Comment by medhatnar** [comment]
> @medhatnar could you include testing instructions for this?

I will include them once Mono pr preview deploys

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.304-alpha.d4a27b.0<br/>@codecademy/gamut-labs@35.11.2-alpha.d4a27b.0<br/>@codecademy/styleguide@64.2.1-alpha.d4a27b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://642f0d91f1fe8200bc9dd1ca--gamut-preview.netlify.app>https://642f0d91f1fe8200bc9dd1ca--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/642f0d91f1fe8200bc9dd1ca>Deploy Logs</a></p>

File: packages/gamut-labs/src/SocialMediaSharing/index.tsx [hunk 4]
```diff
@@ -16,6 +17,12 @@ export type SocialMediaShare = {
   mention?: string;
 };
 
+const UnstyledUnorderedList = styled(GridBox)`
+  list-style: none;
+  padding: 0;
+  margin: 0;
+`;
+
```
Comment (dreamwasp) [inline packages/gamut-labs/src/SocialMediaSharing/index.tsx:25]: the css utility function from gamut-styles would be useful here - docs are [here](https://gamut.codecademy.com/?path=/docs/foundations-system-variants--page) + there are some examples [here](https://www.notion.so/codecademy/Gamut-Best-Practices-ce542cb06ec04e38bf269efab0f6bbb1?pvs=4#b005826dae8a40218881c46547e1c7b6)

**Review by dreamwasp — COMMENTED** [review]
just made a small comment - will do full review when testing instructions are updated ✨

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.306-alpha.44b1e5.0<br/>@codecademy/gamut-labs@35.11.4-alpha.44b1e5.0<br/>@codecademy/styleguide@64.2.3-alpha.44b1e5.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.306-alpha.d89131.0<br/>@codecademy/gamut-labs@35.11.4-alpha.d89131.0<br/>@codecademy/styleguide@64.2.3-alpha.d89131.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://64357f0bab39a822948737a4--gamut-preview.netlify.app>https://64357f0bab39a822948737a4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/64357f0bab39a822948737a4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6435b0bf946ce932d4ab348c--gamut-preview.netlify.app>https://6435b0bf946ce932d4ab348c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6435b0bf946ce932d4ab348c>Deploy Logs</a></p>

**Review by zippyzow — COMMENTED** [review]
Looks like some of the styling got wiped because the border styles are no longer being applied to a Box but instead an Anchor. 

**Prod**
Screenshot 2023-04-11 at 5 28 35 PM

**Monolith PR Env**
Screenshot 2023-04-11 at 5 28 39 PM

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.306-alpha.7753e4.0<br/>@codecademy/gamut-labs@35.11.4-alpha.7753e4.0<br/>@codecademy/styleguide@64.2.3-alpha.7753e4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://643709981be6eb1c17fbed4c--gamut-preview.netlify.app>https://643709981be6eb1c17fbed4c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/643709981be6eb1c17fbed4c>Deploy Logs</a></p>

**Review by saghdaey — APPROVED** [review]

**Review by zippyzow — CHANGES_REQUESTED** [review]
@medhatnar the learner stories and event page look good but the [quiz result page](https://pr-35303-monolith.dev-eks.codecademy.com/explore/sorting-quiz/results?advocator=38&connector=44&questioner=40&solver=37) needs a fix. The 0 margin on the `UnorderedListGridBox` seems to be taking precedence over the margin set in the `socialMediaIcons` css class in the `ArchetypeDetails` component. 

**PR env**
Screenshot 2023-04-12 at 6 27 55 PM
**Prod**
Screenshot 2023-04-12 at 6 27 51 PM

Screenshot 2023-04-12 at 6 33 06 PM

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6437ff7b5e3f6a3772ffc1e0--gamut-preview.netlify.app>https://6437ff7b5e3f6a3772ffc1e0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6437ff7b5e3f6a3772ffc1e0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.306-alpha.89a795.0<br/>@codecademy/gamut-labs@35.11.4-alpha.89a795.0<br/>@codecademy/styleguide@64.2.3-alpha.89a795.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://64380858d316de007fd4e709--gamut-preview.netlify.app>https://64380858d316de007fd4e709--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/64380858d316de007fd4e709>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.306-alpha.e7c6dc.0<br/>@codecademy/gamut-labs@35.11.4-alpha.e7c6dc.0<br/>@codecademy/styleguide@64.2.3-alpha.e7c6dc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6438179e300c83061d2b0924--gamut-preview.netlify.app>https://6438179e300c83061d2b0924--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6438179e300c83061d2b0924>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.306-alpha.8d28ce.0<br/>@codecademy/gamut-labs@35.11.4-alpha.8d28ce.0<br/>@codecademy/styleguide@64.2.3-alpha.8d28ce.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://643822e3986ddf112d09bb27--gamut-preview.netlify.app>https://643822e3986ddf112d09bb27--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/643822e3986ddf112d09bb27>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
nice!

**Review by christian-dinh — APPROVED** [review]

**Pull request merged**